File: Data Structures & Algorithms/is-anagram/submission-1.java
class Solution {
    public boolean isAnagram(String s, String t) {
        if (s.length() != t.length())
        {
            return false;
        }
        else
        {
            int[] alpha = new int[26];
            int len = s.length();
            for(int i=0;i<len;i++)
            {
                alpha[s.charAt(i) - 'a'] ++;
                alpha[t.charAt(i) - 'a'] --;
            }

            for(int num:alpha)
            {
                if(num!=0)
                {
                    return false;
                }
                
            }
        }
        return true;
    }
}

File: Data Structures & Algorithms/anagram-groups/submission-1.cpp
class Solution {
public:
    vector<vector<string>> groupAnagrams(vector<string>& strs) {
        std::unordered_map<string,vector<string>> result;
        for (const auto& word:strs)
        {
            string alpha(26,0);
            for (const auto& i:word)
            {
                alpha[i-'a'] ++;
            }
            result[alpha].push_back(word);
        }
        vector<vector<string>> ans;
        for(const auto& pair:result)
        {
            ans.push_back(pair.second);
        }
        return ans;
    }
};
